### PR TITLE
Fix "Convert to MeshLibrary" not respecting collision transforms

### LINKED ISSUE #56172: GridMap not respecting collision transforms
### Godot version

v4.0.dev.custom_build [489f11ee0], v3.4.stable.official [206ba70f4]

### System information

Windows 10, Vulkan, NVIDIA GeForce RTX 2070 SUPER

### Issue description

Using "Apply MeshInstance Transforms" when exporting a MeshLibrary doesn't apply the transforms to the child collision shapes of the meshes. This results in the collisions being off when used with a grid map.

image

It's possible this was never intended to behave this way given the wording of "Apply MeshInstance Transforms" though I haven't been able to find any documentation to confirm this.

This also happens on 3.4

### Steps to reproduce

1. Make a mesh instance, move its transform off from origin, add a static body + collision shape to the mesh instance
2. Export as mesh library with "Apply MeshInstance Transforms" on
3. Place tiles on a GridMap
4. Turn on "Visible Collision Shapes"
5. Run project and see the collision shapes are not lined up with the meshes

### Minimal reproduction project

[grid_test_v34.zip](https://github.com/godotengine/godot/files/7764615/grid_test_v34.zip)
[gridmap_collision_v4.zip](https://github.com/godotengine/godot/files/7764616/gridmap_collision_v4.zip)

### PR DESCRIPTION
**Pull request by rburing**
When doing Scene → Convert To... → MeshLibrary... there is an option "Apply MeshInstance Transforms" that prevents the zeroing of the transform of each MeshInstance. Unfortunately this option did not respect the transform of the collision shapes of StaticBody children, so effectively the mesh would be moved away from its "left behind" collision shapes. Now the transforms are respected. Fixes https://github.com/godotengine/godot/issues/56172.

Before:

meshlibexport_before

After:

meshlibexport_after

cc @DeleteSystem32

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: editor/plugins/mesh_library_editor_plugin.cpp [hunk 1]
```diff
@@ -136,9 +136,11 @@ void MeshLibraryEditor::_import_scene(Node *p_scene, Ref<MeshLibrary> p_library,
...
 				}
 
-				//Transform3D shape_transform = sb->shape_owner_get_transform(E);
-
-				//shape_transform.set_origin(shape_transform.get_origin() - phys_offset);
+				Transform3D shape_transform;
+				if (p_apply_xforms) {
+					shape_transform = mi->get_transform();
+				}
+				shape_transform *= sb->get_transform() * sb->shape_owner_get_transform(E);
```
Comment (Logharaa) [inline editor/plugins/mesh_library_editor_plugin.cpp:143]: If we don't get into the "if" statement, are you not doing a multiplication with an uninitialised variable?

Comment (rburing) [reply editor/plugins/mesh_library_editor_plugin.cpp:143]: It is not uninitialized: a `Transform3D` is initialized to the identity transform, i.e. with identity basis matrix and zero origin.

Comment (Logharaa) [reply editor/plugins/mesh_library_editor_plugin.cpp:143]: Alright didn't know, thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.

**Pull request merged**